scripts/validate-manifest.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+#
+# validate-manifest.sh [path/to/manifest.json]
+#
+# Dependencies: jq tr sort uniq
+#
+# Description: Ensures consistency of versions specified in manifest.json.
+
+set -eou pipefail
+
+if ! command -v jq > /dev/null; then
+    echo "This script requires jq to be available."
+    exit 1
+fi
+
+GIT_ROOT=$(git rev-parse --show-toplevel)
+MANIFEST_JSON_PATH=${1:-"${GIT_ROOT}/manifest.json"}
+
+declare -a MANIFEST_VERSIONS
+declare -a ROUTE_VERSIONS
+
+# Read the versions array in manifest.json
+readarray -t MANIFEST_VERSIONS < <(jq -r '
+    .versions[]
+    | capture("(?<v>[0-9]+\\.[0-9]+)")
+    | .v' < "${MANIFEST_JSON_PATH}")
+
+# Read all the child paths of changelog/index.md and extract major.minor version
+readarray -t ROUTE_VERSIONS < <(jq -r '
+    .routes[]
+    | select(.path == "./changelog/index.md")
+    | .children[]
+    | .path |
+    capture("(?<v>[0-9]+\\.[0-9]+)")
+    | .v' < "${MANIFEST_JSON_PATH}")
+
+# Compare the two
+DIFF=$(echo "${MANIFEST_VERSIONS[@]}" "${ROUTE_VERSIONS[@]}" | tr ' ' '\n' | sort | uniq -u)
+if [[ -n $DIFF ]]; then
```

Comment:
This seems to be missing an actual `diff` command? I imagine `$DIFF` will always be non-empty.

Reply:
The `uniq -u` does the diff:

```shell
++ echo 1.37 1.36 1.35 1.34 1.33 1.32 1.31 1.30 1.37 1.36 1.35 1.34 1.33 1.32 1.31 1.30
++ tr ' ' '\n'
++ sort
++ uniq -u
+ DIFF=
+ [[ -n '' ]]
+ exit 0
```

Reply:
Ohh, TIL. Nice!